src/Raven.Client/Documents/Operations/DeleteByQueryOperation.cs

```diff
@@ -144,6 +144,23 @@ public override HttpRequestMessage CreateRequest(JsonOperationContext ctx, Serve
                         .Append(_operationId.Value);
                 }
 
+                if (_options.IndexOptions != null && _options.IndexOptions.WaitForIndexes)
+                {
+                    path
+                        .Append("&waitForIndexes=true")
+                        .Append("&waitForIndexesTimeout=")
+                        .Append(_options.IndexOptions.WaitForIndexesTimeout ?? _conventions.WaitForIndexesAfterSaveChangesTimeout)
```

Comment:
you cannot use that timeout from the conventions, it's for session.SaveChanges operation